src/cursor/cur_file.c

```diff
@@ -390,7 +561,22 @@ __curfile_remove(WT_CURSOR *cursor)
     WT_DECL_RET;
     WT_SESSION_IMPL *session;
     uint64_t time_start, time_stop;
-    bool positioned;
+    bool positioned, released;
+
+    positioned = released = false;
+    cbt = (WT_CURSOR_BTREE *)cursor;
+    CURSOR_REMOVE_API_CALL(cursor, session, CUR2BT(cbt));
+
+    if (F_ISSET(cbt, WT_CBT_REPOSITION)) {
+        WT_ERR_NOTFOUND_OK(__curfile_reposition(cursor, true, true, NULL), true);
+        if (ret == WT_NOTFOUND) {
+            __wt_verbose_notice(session, WT_VERB_ERROR_RETURNS, "%s",
+              "WT_ROLLBACK: rolling back cursor remove as initial position was lost");
+            WT_ERR(WT_ROLLBACK);
```

Comment:
I'm confused by this code. Why isn't this a panic? If there's a "normal" case where reposition fails, we can't do that, it's not OK for us to lose a cursor's position under pretty much any circumstances.

Reply:
It is possible the key is just removed by other cursor, which is not a fatal error. With WT-8461, we can just return WT_NOT_FOUND.